### PR TITLE
change legacy metadata fileRelativePath outputRootRelativePath resolve logic

### PR DESCRIPTION
**Pull request by 928PJY**

### CODE REVIEW

File: src/docfx/build/legacy/LegacyUtility.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ public static string ToLegacySiteUrlRelativeToBaseSitePath(this Document doc, Do
         }
 
         public static string ToLegacyOutputPath(this LegacyManifestOutputItem legacyManifestOutputItem, Docset docset, string groupId)
-            => Path.Combine($"{groupId}", docset.Config.DocumentId.SiteBasePath, legacyManifestOutputItem.RelativePath);
+            => Path.Combine(docset.Config.DocumentId.SiteBasePath, $"{groupId}", legacyManifestOutputItem.RelativePath);
```
Comment (superyyrrzz) [inline src/docfx/build/legacy/LegacyUtility.cs]: >$"{groupId}" [](start = 67, length = 12)

directly `groupId`?

File: src/docfx/build/moniker/MonikersProvider.cs [hunk 2]
```diff
@@ -96,5 +96,17 @@ public List<string> GetZoneMonikers(string rangeString, List<string> fileLevelMo
             monikers.Sort(Comparer);
             return monikers;
         }
+
+        public string GetGroupIdFromMonikerList(IEnumerable<string> monikers)
```
Comment (superyyrrzz) [inline src/docfx/build/moniker/MonikersProvider.cs]: `IEnumerable` => `List`?

File: src/docfx/build/moniker/MonikersProvider.cs [hunk 3]
```diff
@@ -96,5 +96,17 @@ public List<string> GetZoneMonikers(string rangeString, List<string> fileLevelMo
...
+        {
+            if (monikers == null || monikers.Count() == 0)
+            {
+                return null;
+            }
+            var sortedMonikers = monikers.ToList();
+            sortedMonikers.Sort(Comparer);
```
Comment (superyyrrzz) [inline src/docfx/build/moniker/MonikersProvider.cs]: Seems there are many duplicates `Sort` here. Add some cache?

File: src/docfx/build/moniker/MonikersProvider.cs (hunk 2 above)
Comment (superyyrrzz) [reply src/docfx/build/moniker/MonikersProvider.cs]: What about `string HashUtility.GetMd5HashShortStable(List<string> items)`?

File: src/docfx/build/legacy/LegacyUtility.cs (hunk 1 above)
Comment (928PJY) [reply src/docfx/build/legacy/LegacyUtility.cs]: groupId maybe null

---
In reply to: [240446206](https://github.com/dotnet/docfx/pull/3750#discussion_r240446206) [](ancestors = 240446206)

File: src/docfx/build/moniker/MonikersProvider.cs (hunk 3 above)
Comment (928PJY) [reply src/docfx/build/moniker/MonikersProvider.cs]: yes, but this is a utility, we have to make sure the output is stable

---
In reply to: [240448390](https://github.com/dotnet/docfx/pull/3750#discussion_r240448390) [](ancestors = 240448390)

File: src/docfx/build/moniker/MonikersProvider.cs (hunk 2 above)
Comment (928PJY) [reply src/docfx/build/moniker/MonikersProvider.cs]: that requires a comparer to be parsed into?

---
In reply to: [240448635](https://github.com/dotnet/docfx/pull/3750#discussion_r240448635) [](ancestors = 240448635)

File: src/docfx/lib/HashUtility.cs [hunk 4]
```diff
@@ -54,6 +55,16 @@ public static string GetFileSha1Hash(string fileName)
             }
         }
 
+        public static string GetSortedMonikerListHash(List<string> monikers)
```
Comment (superyyrrzz) [inline src/docfx/lib/HashUtility.cs]: naming: `public static string GetMd5HashShort(List<string> items)`. Feels it more suitable in `HashUtility`

File: src/docfx/lib/HashUtility.cs [hunk 5]
```diff
@@ -54,6 +55,16 @@ public static string GetFileSha1Hash(string fileName)
...
+        {
+            if (monikers == null || monikers.Count == 0)
```
Comment (live1206) [inline src/docfx/lib/HashUtility.cs]: Why do we want to put a `moninker` specific method in `HashUtility`? And do we really need another method for the null check?

**Pull request merged**